tests/integration/components/bs-button-test.js

```diff
@@ -9,11 +9,44 @@ import hbs from 'htmlbars-inline-precompile';
 import setupNoDeprecations from '../../helpers/setup-no-deprecations';
 import a11yAudit from 'ember-a11y-testing/test-support/audit';
 import sinon from 'sinon';
+/* global QUnit */
 
 module('Integration | Component | bs-button', function (hooks) {
   setupRenderingTest(hooks);
   setupNoDeprecations(hooks);
 
+  hooks.beforeEach(function (assert) {
+    /*
+     * The custom assertions `rejectsErrorNotGoinghThroughEmberOnError` allows to test
+     * functionality, which rejects intentionally. It serves a similar use case as
+     * `assert.throws` and `setupOnerror` utility provided by `@ember/test-helpers`.
+     * Sadly for some use cases none of them is working.
+     *
+     * `rejectsErrorNotGoinghThroughEmberOnError` should only be used for cases in which
+     * neither  `assert.throws` nor `setupOnerror` could be used. Otherwise these two
+     * methods to test rejecting promises should be preferred.
+     */
+    assert.rejectsErrorNotGoingThroughEmberOnError = async function (callback, expectedError) {
```

Comment:
Should we put this into our custom test helpers module, so we can reuse it in the future?